src/pp-annotator/form/AnnotationForm.tsx

```diff
@@ -54,77 +63,187 @@ export default class AnnotationForm extends React.Component<
 
     this.handleInputChange = this.handleInputChange.bind(this);
     this.onSave = this.onSave.bind(this);
+    this.executeSave = this.executeSave.bind(this);
     this.onCancel = this.onCancel.bind(this);
   }
 
+  public componentWillUpdate(_nextProps: IAnnotationFormProps, nextState: Partial<IAnnotationFormState>) {
+    // Whenever the field has changed, eradicate the error message
+    if(nextState.referenceLink) {
+      nextState.referenceLinkError = '';
+    }
+    if(nextState.referenceLinkTitle) {
+      nextState.referenceLinkTitleError = '';
+    }
+  }
+
+  public saveButtonClass() {
+    const priorityToClass = {
```

Comment:
I'd suggest moving this as a class property(like `noCommentModal`) , so it's not created every time `saveButtonClass` is called

Reply:
done